ShowdownAI.Middleware/ShowdownAI.Middleware/Services/MoveDataLookup.cs

```diff
@@ -8,37 +8,44 @@ namespace ShowdownAI.Middleware.Services
     public class MoveDataLookup
     {
         private const string _moveDataPath = "./MoveData/move_data.json";
-        private readonly string _jsonData;    
+        private readonly string _jsonData;
+
+        private Dictionary<string, MoveData> _moveData = new();
 
         public MoveDataLookup()
         {
             _jsonData = File.ReadAllText(_moveDataPath);
+            GenerateMoveData();
         }
 
-        public MoveData GetMoveData(string moveId)
+        public void GenerateMoveData()
         {
-            MoveData moveData = new() { Id = moveId };
-
             using (JsonDocument document = JsonDocument.Parse(_jsonData))
             {
                 JsonElement moves = document.RootElement.GetProperty("moves");
 
-                foreach (JsonElement move in moves.EnumerateArray())
+                foreach (JsonElement moveObject in moves.EnumerateArray())
                 {
-                    if (move.TryGetProperty(moveId, out JsonElement moveAttributes))
+                    JsonProperty move = moveObject.EnumerateObject().FirstOrDefault();
+
+                    MoveData moveData = new() { Id = move.Name };
+
+                    foreach (JsonProperty moveAttribute in move.Value.EnumerateObject())
                     {
-                        foreach (JsonProperty moveAttribute in moveAttributes.EnumerateObject())
-                        {
-                            AddAttributeToMoveData(moveAttribute, moveData);
-                        }
+                        AddAttributeToMoveData(moveAttribute, moveData);
                     }
+
+                    _moveData.Add(move.Name, moveData);
                 }
             }
+        }
 
-            return moveData;
+        public MoveData GetMoveData(string moveId)
+        {
+            return _moveData[moveId];
```

Comment:
Can you replace this with a TryGetValue in case there's an issue with a specific move so it won't crash